### ISSUE TITLE
CLI::ExistingDirectory falsely failing with unicode file path

### ISSUE THREAD

**Issue by BullyWiiPlaza** [henryiii/CLI11#1065]
Hello,

I've been trying to parse a unicode file path using `CLI11` on Windows. First, I switched my main method's signature to the wide string version: `int wmain(const int argument_count, wchar_t* arguments[])`

Then I defined a `std::filesystem::path` object for holding the file path:

```
std::filesystem::path source_directory;
application.add_option("--source-directory", source_directory, "The source directory to use")
    ->required()
    ->check(CLI::ExistingDirectory);
```

Finally, I parsed my command line options using the `CLI11` macro:

`CLI11_PARSE(application, argument_count, arguments)`

This code works fine with ASCII only file paths but if I run it with e.g. a Unicode file path (Arabic symbols included), CLI11 will throw an exception as follows:
```
--source-directory: Directory does not exist: xxx
Run with --help for more information.
```

This seems to be a bug because I made sure the file path is correct, I can work with the file path without any problems with the `std::filesystem::path` libraries. Also, a manually added

```
if (!is_directory(source_directory))
{
    throw std::runtime_error("Source directory isn't a directory");
}
```

after parsing using `CLI11` and after dropping the directory existence check `->check(CLI::ExistingDirectory)`, it does **NOT** trigger the `std::runtime_error` since the file path indeed exists but `CLI11` claims otherwise.

Why does `CLI11`'s `CLI::ExistingDirectory` not work as expected with unicode file paths? Is this a bug or am I still likely doing something wrong here?

---

After reading the source code a bit, it seems like CLI11 calls `.c_str()` on the file path before validating which might turn out wrong here:
https://github.com/CLIUtils/CLI11/blob/f4f225d9a233583604fc1bf6e8d1fa0ccd60c652/include/CLI/impl/Validators_inl.hpp#L185

The right way would be to call `.wstring()` on the `std::filesystem::path` and then using the wide string file path APIs or `std::filesystem::path` directly. However, CLI11 aims to be a C++11 library still, right? 😮

**Comment by phlptp** [comment]
Could you add a failing test and a make PR for this issue?   I suspect part of the issue is that none of the main developers work with wide strings on a regular basis.   So if there was a test we could work with that would help a lot in resolving it.

**Comment by BullyWiiPlaza** [comment]
Oops, after writing the test, I realized that `CLI11` was working correctly since `check_path()` called `std::filesystem::status`. I tried my application again and it still failed. So where did the issue come from? I used the package manager `vcpkg` to install `CLI11` and upon inspecting the build definition, it said this:
```
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
        -DCLI11_BUILD_EXAMPLES=OFF
        -DCLI11_BUILD_DOCS=OFF
        -DCLI11_BUILD_TESTS=OFF
        -DCLI11_PRECOMPILED=ON
)
```
So they're use pre-compilation for `CLI11`. After loading the `CLI11.lib` file and disassembling the `check_path()` function, it showed that the OS specific API was compiled despite the fact that I'm using `C++17` and newer which would support `std::filsystem`.

If I disable pre-compiling, the `std::filesystem` API is available and called, all good from here. Now another question arises if the vcpkg port should use pre-compilation turned off or if we can somehow instruct the pre-compilation to include the `std::fileystem` API with the macro `CLI11_HAS_FILESYSTEM` being set.

So overall, the best I could come up with is forcing `C++17` for the `vcpkg` port compilation, then everything is fine:
```
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
        -DCLI11_BUILD_EXAMPLES=OFF
        -DCLI11_BUILD_DOCS=OFF
        -DCLI11_BUILD_TESTS=OFF
        -DCLI11_PRECOMPILED=ON
    CMAKE_OPTIONS
        -DCMAKE_CXX_STANDARD=17
)
```
However, since the problem wasn't technically caused by `CLI11`, I'm now closing this issue.
